lru/__init__.py

```diff
@@ -179,25 +190,35 @@ def cleanup(self):
         if self.expiration is None:
             return None
         t = int(time.time())
-        #Delete expired
+        # Delete expired
+        keys_to_delete = []
         next_expire = None
         for k in self.__expire_times:
             if self.__expire_times[k] < t:
-                self.__delete__(k)
+                keys_to_delete.append(k)
             else:
                 next_expire = self.__expire_times[k]
                 break
 
-        #If we have more than self.max_size items, delete the oldest
-        while (len(self.__values) > self.max_size):
+        for k in keys_to_delete:
+            self.__delete__(k)
+
+        keys_to_delete = []
+        # If we have more than self.max_size items, delete the oldest
+        while len(self.__values) > self.max_size:
             for k in self.__access_times:
-                self.__delete__(k)
+                keys_to_delete.append(k)
                 break
+
+        for k in keys_to_delete:
```

Comment:
This section should probably go inside the `while` loop. Please refer to the new #18.